### ISSUE TITLE
re: Android file path storage for assets--

### ISSUE THREAD

**Issue by zeukterialpneumaion** [yvanvds/yse-soundengine#17]
## Awesome library ...thankyou!
am prototyping some realtime synthesis for my android app at the moment ... am using android studio and kotlin ... however, it would be very useful to have access to files ... want to do some dsp on incoming microphone streams and some loop sampling ...
...have you seen this thread?
https://stackoverflow.com/questions/6346889/how-to-reference-an-asset-in-a-library-project
and this::
https://github.com/donkirkby/webandnative/blob/assets/android/.project#L33

**Comment by yvanvds** [comment]
I agree, but it's not that simple. It's one of the reasons why JUCE needs its java hooks. I'm sure I'll look into this sooner or later, but I cannot work full-time on this library and this functionality not a number one priority for me.

If you need to load small files you should look at the BufferIO class. (There is also a demo about it somewhere.) With this class you can pass raw memory buffers (which can be loaded from disk in your java application) to the engine. 

It is not very efficient because this means copying the entire buffer when passing it to the native side, but I see this as a temporary solution, workable for small files.

**Comment by yvanvds** [comment]
Yes, but it would be a partial solution only. You still have to pass the jnienv variable from the java side to get it to work. That's fine for some applications, but it won't work with Xamarin, for example. 

On the other hand, a partial solution might still be an improvement. I'll think about it.

Right now I am working on a build system for android studio. It might be a good alternative if you don't use Visual Studio and help you to implement an asset manager. It's mostly ready, but there are some quirks i have to sort out before I push to github.

**Comment by yvanvds** [comment]
Hi @zeukterialpneumaion,

Apologies for the very long silence on this — the project went on hiatus and is only being maintained again recently. v2.1.0 just shipped.

The big news for you: **there's a real Android backend now** (NDK + CMake + Oboe + NativeActivity demo APK), so the engine actually runs on device. Microphone/loop sampling-style projects are feasible.

What's *not* solved out of the box is asset access through `AAssetManager` — that still has to be wired up by the app since it depends on each project's Java/Kotlin glue. The C API has hooks for custom file readers if you want to plug `AAssetManager` into the engine's I/O path. If you want to try it and run into integration issues, please open a fresh issue.

Closing this one for hygiene. Sorry again for the wait.
